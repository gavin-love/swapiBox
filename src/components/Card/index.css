.card {
  border-top: 2px solid sandybrown;
}
