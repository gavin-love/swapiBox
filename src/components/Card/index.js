import React from "react";

const Card = props => {
  if (props.homeworld) {
    return (
      <div>
        <h1>{props.name}</h1>
        <h1>{props.homeworld}</h1>
        <h1>{props.species}</h1>
        <h1>{props.population}</h1>
        <button onClick={() => props.handleFavorites(props)}>favorite</button>
      </div>
    );
  }

  if (props.terrain) {
    return (
      <div>
        <h1>{props.name}</h1>
        <h1>{props.terrain}</h1>
        <h1>{props.population}</h1>
        <h1>{props.climate}</h1>
        <h1>{props.residents.map(resident => resident.name)}</h1>
        <button onClick={() => props.handleFavorites(props)}>favorite</button>
      </div>
    );
  }
  if (props.passengers) {
    return (
      <div>
        <h1>{props.name}</h1>
        <h1>{props.model}</h1>
        <h1>{props.class}</h1>
        <h1>{props.passengers}</h1>
        <button onClick={() => props.handleFavorites(props)}>favorite</button>
      </div>
    );
  }
  return <div />;
};

export default Card;
