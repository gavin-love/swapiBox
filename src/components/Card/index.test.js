import React from "react";
import Card from "./index.js";
import { shallow } from "enzyme";

describe("Card", () => {
  let wrapper;
  let mockData;

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should return a people card when passed an object with property of homeworld", () => {
    mockData = {
      name: "Luke Skywalker",
      homeworld: "Tatooine",
      species: "Human",
      population: "200000"
    };
    wrapper = shallow(
      <Card
        name={mockData.name}
        homeworld={mockData.homeworld}
        species={mockData.species}
        population={mockData.population}
        key={"420"}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should return a planets card when passed an object with a property of terrain", () => {
    mockData = {
      name: "Alderaan",
      terrain: "grasslands, mountains",
      population: "2000000000",
      climate: "temperate",
      residents: ["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"]
    };
    wrapper = shallow(
      <Card
        name={mockData.name}
        terrain={mockData.terrain}
        population={mockData.population}
        climate={mockData.climate}
        residents={mockData.residents}
        key={"420"}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should return a vehicle card when passed an object with a property of passengers", () => {
    mockData = {
      name: "Sand Crawler",
      model: "Digger Crawler",
      class: "wheeled",
      passengers: "30"
    };

    wrapper = shallow(
      <Card
        name={mockData.name}
        model={mockData.model}
        class={mockData.class}
        passengers={mockData.passengers}
        key={"420"}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
