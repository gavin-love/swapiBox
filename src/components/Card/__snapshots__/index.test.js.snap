// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `undefined`;

exports[`Card should return a people card when passed an object with property of homeworld 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    homeworld="Tatooine"
    name="Luke Skywalker"
    population="200000"
    species="Human"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Luke Skywalker
</h1>,
        <h1>
          Tatooine
</h1>,
        <h1>
          Human
</h1>,
        <h1>
          200000
</h1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tatooine",
        },
        "ref": null,
        "rendered": "Tatooine",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Human",
        },
        "ref": null,
        "rendered": "Human",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "200000",
        },
        "ref": null,
        "rendered": "200000",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Luke Skywalker
</h1>,
          <h1>
            Tatooine
</h1>,
          <h1>
            Human
</h1>,
          <h1>
            200000
</h1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tatooine",
          },
          "ref": null,
          "rendered": "Tatooine",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Human",
          },
          "ref": null,
          "rendered": "Human",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "200000",
          },
          "ref": null,
          "rendered": "200000",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should return a planets card when passed an object with a property of terrain 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    climate="temperate"
    name="Alderaan"
    population="2000000000"
    residents={
        Array [
            "Leia Organa",
            "Bail Prestor Organa",
            "Raymus Antilles",
          ]
    }
    terrain="grasslands, mountains"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Alderaan
</h1>,
        <h1>
          grasslands, mountains
</h1>,
        <h1>
          2000000000
</h1>,
        <h1>
          temperate
</h1>,
        <h1>
          
</h1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Alderaan",
        },
        "ref": null,
        "rendered": "Alderaan",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "grasslands, mountains",
        },
        "ref": null,
        "rendered": "grasslands, mountains",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2000000000",
        },
        "ref": null,
        "rendered": "2000000000",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "temperate",
        },
        "ref": null,
        "rendered": "temperate",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
          undefined,
        ],
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Alderaan
</h1>,
          <h1>
            grasslands, mountains
</h1>,
          <h1>
            2000000000
</h1>,
          <h1>
            temperate
</h1>,
          <h1>
            
</h1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alderaan",
          },
          "ref": null,
          "rendered": "Alderaan",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "grasslands, mountains",
          },
          "ref": null,
          "rendered": "grasslands, mountains",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2000000000",
          },
          "ref": null,
          "rendered": "2000000000",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "temperate",
          },
          "ref": null,
          "rendered": "temperate",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
            undefined,
          ],
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should return a vehicle card when passed an object with a property of passengers 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    class="wheeled"
    model="Digger Crawler"
    name="Sand Crawler"
    passengers="30"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Sand Crawler
</h1>,
        <h1>
          Digger Crawler
</h1>,
        <h1>
          wheeled
</h1>,
        <h1>
          30
</h1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sand Crawler",
        },
        "ref": null,
        "rendered": "Sand Crawler",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Digger Crawler",
        },
        "ref": null,
        "rendered": "Digger Crawler",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "wheeled",
        },
        "ref": null,
        "rendered": "wheeled",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "30",
        },
        "ref": null,
        "rendered": "30",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Sand Crawler
</h1>,
          <h1>
            Digger Crawler
</h1>,
          <h1>
            wheeled
</h1>,
          <h1>
            30
</h1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sand Crawler",
          },
          "ref": null,
          "rendered": "Sand Crawler",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Digger Crawler",
          },
          "ref": null,
          "rendered": "Digger Crawler",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "wheeled",
          },
          "ref": null,
          "rendered": "wheeled",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "30",
          },
          "ref": null,
          "rendered": "30",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
